Wrong kafka source configuration results in SIGSEGV instead of useful error message

<!-- 
Are you using Knative? If you do, we would love to know!
https://github.com/knative/community/issues/new?template=ADOPTERS.yaml&title=%5BADOPTERS%5D%3A+%24%7BCOMPANY+NAME+HERE%7D
-->

**Describe the bug**
Wrong kafka source configuration results in SIGSEGV instead of useful error message.

 **Expected behavior**
There should be a clear error message instead of a SIGSEGV.

 **To Reproduce**
Use `plain` instead of `PLAIN` for `net.sasl.type`.

 **Knative release version**
1.14.5

 **Additional context**
Log output:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c6f119]

goroutine 532 [running]:
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).reconcileInitialOffset(0xc000142000, {0x261e1d8, 0xc006ea7c20}, 0xc00068b508)
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:599 +0x2d9
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).ReconcileKind(0xc000142000, {0x261e1d8, 0xc006ea7c20}, 0xc00068b508)
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:196 +0x131
knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup.(*reconcilerImpl).Reconcile(0xc0004aa5a0, {0x261e1d8, 0xc006ea7bf0}, {0xc006eae4c0, 0x31})
knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup/reconciler.go:240 +0x52b
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc0001b3980)
knative.dev/pkg@v0.0.0-20240416145024-0f34a8815650/controller/controller.go:542 +0x508
knative.dev/pkg/controller.(*Impl).RunContext.func3()
knative.dev/pkg@v0.0.0-20240416145024-0f34a8815650/controller/controller.go:491 +0x59
created by knative.dev/pkg/controller.(*Impl).RunContext in goroutine 506
knative.dev/pkg@v0.0.0-20240416145024-0f34a8815650/controller/controller.go:489 +0x325
```

I think this is cause by calling `defer kafkaClusterAdminClient.Close()` even if no client was created (https://github.com/knative-extensions/eventing-kafka-broker/blob/main/control-plane/pkg/reconciler/consumergroup/consumergroup.go#L602).

--- a/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_validation.go
+++ b/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_validation.go
@@ -57,6 +57,9 @@ func (cts *ConsumerTemplateSpec) Validate(ctx context.Context) *apis.FieldError 
 		cts.Spec.Delivery.Validate(specCtx).ViaField("delivery"),
 		cts.Spec.Subscriber.Validate(specCtx).ViaField("subscriber"),
 	)
+	if cts.Spec.Configs.Configs == nil {
+		err = err.Also(apis.ErrMissingField("spec.configs"))
+	}
 	return err
 }
 

--- a/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_validation_test.go
+++ b/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_group_validation_test.go
@@ -109,6 +109,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Host:   "127.0.0.1",
 								},
 							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
+							},
 						},
 					},
 				},
@@ -135,6 +138,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Name:       "my-seq",
 									APIVersion: "flows.knative.dev/v1",
 								},
+							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
 							},
 						},
 					},
@@ -191,6 +197,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Host:   "127.0.0.1",
 								},
 							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
+							},
 						},
 					},
 				},
@@ -214,6 +223,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Scheme: "http",
 									Host:   "127.0.0.1",
 								},
+							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
 							},
 						},
 					},
@@ -243,6 +255,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Host:   "127.0.0.1",
 								},
 							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
+							},
 						},
 					},
 				},
@@ -266,6 +281,9 @@ func TestConsumerGroup_Validate(t *testing.T) {
 									Scheme: "http",
 									Host:   "127.0.0.1",
 								},
+							},
+							Configs: ConsumerConfigs{
+								Configs: map[string]string{},
 							},
 						},
 					},

--- a/control-plane/pkg/reconciler/consumergroup/consumergroup.go
+++ b/control-plane/pkg/reconciler/consumergroup/consumergroup.go
@@ -301,13 +301,17 @@ func (r *Reconciler) deleteConsumerGroupMetadata(ctx context.Context, cg *kafkai
 		return fmt.Errorf("failed to get secret for Kafka cluster auth: %w", err)
 	}
 
+	if cg.Spec.Template.Spec.Configs.Configs == nil {
+		return fmt.Errorf("no consumer config supplied, unable to get bootstrap.servers")
+	}
+
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])
 
 	kafkaClusterAdminClient, err := r.GetKafkaClusterAdmin(ctx, bootstrapServers, kafakSecret)
-	defer kafkaClusterAdminClient.Close()
 	if err != nil {
 		return fmt.Errorf("cannot obtain Kafka cluster admin, %w", err)
 	}
+	defer kafkaClusterAdminClient.Close()
 
 	groupId := cg.Spec.Template.Spec.Configs.Configs["group.id"]
 	if err = kafkaClusterAdminClient.DeleteConsumerGroup(groupId); err != nil && !errorIsOneOf(err, sarama.ErrUnknownTopicOrPartition, sarama.ErrGroupIDNotFound) {
@@ -596,13 +600,17 @@ func (r *Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkaintern
 		return fmt.Errorf("failed to get secret for Kafka cluster auth: %w", err)
 	}
 
+	if cg.Spec.Template.Spec.Configs.Configs == nil {
+		return fmt.Errorf("no consumer config supplied, unable to get bootstrap.servers")
+	}
+
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])
 
 	kafkaClusterAdminClient, err := r.GetKafkaClusterAdmin(ctx, bootstrapServers, kafkaSecret)
-	defer kafkaClusterAdminClient.Close()
 	if err != nil {
 		return fmt.Errorf("cannot obtain Kafka cluster admin, %w", err)
 	}
+	defer kafkaClusterAdminClient.Close()
 
 	kafkaClient, err := r.GetKafkaClient(ctx, bootstrapServers, kafkaSecret)
 	if err != nil {

--- a/control-plane/pkg/reconciler/consumergroup/consumergroup.go
+++ b/control-plane/pkg/reconciler/consumergroup/consumergroup.go
@@ -301,10 +301,6 @@ func (r *Reconciler) deleteConsumerGroupMetadata(ctx context.Context, cg *kafkai
 		return fmt.Errorf("failed to get secret for Kafka cluster auth: %w", err)
 	}
 
-	if cg.Spec.Template.Spec.Configs.Configs == nil {
-		return fmt.Errorf("no consumer config supplied, unable to get bootstrap.servers")
-	}
-
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])
 
 	kafkaClusterAdminClient, err := r.GetKafkaClusterAdmin(ctx, bootstrapServers, kafakSecret)
@@ -598,10 +594,6 @@ func (r *Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkaintern
 	kafkaSecret, err := r.newAuthSecret(ctx, cg)
 	if err != nil {
 		return fmt.Errorf("failed to get secret for Kafka cluster auth: %w", err)
-	}
-
-	if cg.Spec.Template.Spec.Configs.Configs == nil {
-		return fmt.Errorf("no consumer config supplied, unable to get bootstrap.servers")
 	}
 
 	bootstrapServers := kafka.BootstrapServersArray(cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"])

